packages/images/docs/find-by-image.md

```diff
@@ -0,0 +1,65 @@
+## Finding and Interacting with Image Elements
+
+Using the `-image` locator strategy supported by this plugin, it is possible to send an Appium an image file representing an element you want to tap. If Appium can find a screen region matching your template, it will wrap up information about this region as a standard `WebElement` and send it back to your Appium client.
+
+The strategy will be made available differently for each Appium client, for example: `driver.findElementByImage()`.
+
+### Image Selectors
+
+In conjunction with any locator strategy, you need to use a "selector" which details the specific nature of your find request. In the case of the `-image` strategy, the selector must be a string which is a base64-encoded image file representing the template you want to use for matching.
+
+### Image Elements
+
+If the image match is successful, Appium will cache information about the match and create a standard response for your client to consume, resulting in the instantiation of a standard element object in your test script. Using this element object, you are able to call a small number of methods on the "Image Element", as if it were a bona-fide `WebElement`:
+
+* `click`
+* `isDisplayed`
+* `getSize`
+* `getLocation`
+* `getLocationInView`
+* `getElementRect`
+* `getAttribute`
+    * `visual` returns matched image as base64 data if `getMatchedImageResult` is `true`
+    * `score` returns the similarity score as a float number in range `[0.0, 1.0]` sine Appium 1.18.0
+
+These actions are supported on "Image Elements" because they are the actions which involve only use of screen position for their functioning. Other actions (like `sendKeys`, for example) are not supported, because all Appium can know based on your template image is whether or not there is a screen region which visually matches it--Appium has no way of turning that information into a driver-specific UI element object, which would be necessary for the use of other actions.
+
+It's important to keep this important point in mind: there is nothing "magic" about Image Elements---they merely reference screen coordinates, and thus "tapping" an Image Element is internally nothing more than Appium constructing a tap at a point in the center of the Image Element's screen bounds (and in fact you can tell Appium which API to use to perform that tap---see below).
+
+### Related Settings
+
+Because finding elements by image relies on image analysis software in conjunction with Appium's screenshot functionality and the reference images you yourself provide, we provide a number of settings that help you modulate this feature, in some cases potentially speeding up the match or making it more accurate.
+
+To access these settings, you should use the Appium [Settings API](https://github.com/appium/appium/blob/master/docs/en/advanced-concepts/settings.md). These are the settings that are available:
```

Comment:
It would be nice to address a link to https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#update-settings like:

```
The Settings API can set as part of capabilities as [this](https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#update-settings ) so that you can include them in capabilities.
```

Internally, Appium calls the settings endpoint so this also should work in plugins.

Reply:
good idea, i've done that now.